src/browser/progress-bar.js

```diff
@@ -6,164 +6,125 @@ define((require, exports, module) => {
 
-  const style = StyleSheet.create({
+  const barStyle = StyleSheet.create({
     base: {
-      zIndex: 1,
-      display: 'block',
-      width: '100%',
       height: 4,
-      marginLeft: '-100%',
       position: 'absolute',
-      top: 28,
+      top: 0,
       left: 0,
-      backgroundColor: null,
-      opacity: null,
-      transform: null
+      width: '100%',
+      transitionProperty: 'transform, opacity',
+      transitionDuration: '300ms, 200ms',
+      transitionTimingFunction: 'ease-out',
+      transitionDelay: '0s, 300ms',
+
+      animationDuration: '10s',
+      animationTimingFunction: 'cubic-bezier(0, 0.5, 0, 0.5)',
+
+      transform: 'translateX(-100vw)',
+      opacity: 0,
+    },
+    hidden: {
+      visibility: 'hidden'
+    },
+    visible: {
+      visibility: 'visible'
+    },
+    loading: {
+      animationName: 'progressBarLoading'
+    },
+    loaded: {
+      transform: 'translateX(0vw)',
+      opacity: 0
     }
   });
 
-  // View
+  const arrowStyle = StyleSheet.create({
+    base: {
+      width: 4,
+      height: 4,
+      float: 'right',
+      marginRight: -4
+    }
+  });
 
-  const startFading = 0.8;    // When does opacity starts decreasing to 0
-  const computeOpacity = progress =>
-    progress < startFading ? 1 :
-    1 - Math.pow((progress - startFading) / (1 - startFading), 1);
-
-  const ProgressUpdate = (id, {timeStamp}) =>
-    LoadProgress({id, timeStamp});
-
-  const view = (id, progress, theme, address) => {
-    const node = progress && html.div({
-      key: 'ProgressBar',
-      style: Style(style.base, {
-        backgroundColor: theme.progressBar,
-        opacity: computeOpacity(progress.value),
-        transform: `translateX(${100 * progress.value}%)`
+  const progressbarView = (id, progress, isSelected, theme) => {
+    return html.div({
+      key: `progressbar-${id}`,
+      style: Style(barStyle.base,
+                   !isSelected ? barStyle.hidden : barStyle.visible,
+                   progress.loadEnded ? barStyle.loaded : barStyle.loading, {
+                     backgroundColor: theme.progressBar
+                   }),
+    }, [html.div({
+      key: `progressbar-arrow-${id}`,
+      style: Style(arrowStyle.base, {
+        backgroundImage: `linear-gradient(135deg, ${theme.progressBar} 50%, transparent 50%)`,
       })
-    });
+    })]);
+  };
+
+  const containerStyle = StyleSheet.create({
+    base: {
+      position: 'absolute',
+      top: 28,
+      left: 0,
+      width: '100vw',
+      height: 4,
+      overflow: 'hidden',
+      pointerEvents: 'none',
+    }
+  });
 
-    // If `webView` is loading then animate node with `ProgressChange` actions
-    // on every animatation frame.
-    return !progress ? null :
-           !isLoading(progress) ? node :
-           animate(node, address.pass(ProgressUpdate, id));
+  const view = (loaders, progress, selected, theme) => {
+    return html.div({
+      key: 'progressbars',
+      style: Style(containerStyle.base)
```

Comment:
Nit: No need to wrap single item in `Style` it's just for composing several of them.